Improved concepts (#1051)
casper-network/docs: source/docs/casper/concepts/index.md

---
title: Casper Concepts
---
## Introduction

> This section is intended for the developer or user who has had little or no exposure to Casper and wants to understand the concepts underpinning the platform. Working with the Casper blockchain requires an understanding of some core concepts. Some of these concepts generally apply to blockchain technology, while others are Casper-specific. If you are new to Casper, we recommend you familiarize yourself with these concepts to ensure clarity at a further stage of your educative process.

Reviewer: ```suggestion
This section is intended for the developer or user who has had little or no exposure to Casper and wants to understand the concepts underpinning the platform. Working with the Casper blockchain requires an understanding of some core concepts. These concepts generally apply to blockchain technology, although some are Casper-specific. If you are new to Casper, we recommend you familiarize yourself with these concepts to ensure a proper foundation for your Casper education.
```

Once you understand the basic concepts underpinning Casper, here are some further learning paths you may wish to pursue:

 - If you are just getting started and looking for help building your first Casper-based app, we recommend the [beginner tutorials](../resources/tutorials/beginner/index.md) section.
 - If you have already started your Casper education and want to continue with more advanced tutorials, try our [advanced tutorials](../resources/tutorials/advanced/index.md) section.
 - If you already have experience with or know of the Casper platform and are looking for specific information about a development topic, head over to the [development](../developers/index.md) area.

## Casper Concepts

### [Accounts and Cryptographic Keys](./accounts-and-keys.md)

The Casper programming model is account-based. An understanding of how Casper accounts work, and how they are secured, may be read [here](./accounts-and-keys.md).

### [Hash Types](./hash-types.md)

> Hashes are used everywhere in Casper for keys, addresses, packaging data nd more. Find a description of these hash types [here](./hash-types.md).

Reviewer: ```suggestion
Hashes are used throughout the Casper ecosystem for keys, addresses, packaging data and more. Find a description of these hash types [here](./hash-types.md).
```

### [Call Stacks](./callstack.md)

[This document](./callstack.md) contains a discussion of how Casper manages the calling of a contract, and some essential concepts to understand when running this kind of code.

### [Contracts and Session Code](./session-code.md)

Read [this](./session-code.md) to understand the different kinds of code executed on the Casper platform, their differences, and why you would use one versus the other.

### [Dictionaries](./dictionaries.md)

[Dictionaries](./dictionaries.md) are a primary construct for storing and retrieving data on the Casper platform.

### [Design](./design/index.md)

> An in-depth look at the high-level design and architecture of the Casper eco-system can be found [here](./design/index.md).

Reviewer: ```suggestion
An in-depth look at the high-level design and architecture of the Casper ecosystem can be found [here](./design/index.md).
```
